sched/scheduler/stateful_scheduler.go

```diff
@@ -116,15 +113,16 @@ func NewStatefulScheduler(
 		}()
 	}
 
+	// Recover Jobs in a separate go routine to allow the scheduler
```

Comment:
I wonder if we will want some way to signal that recovery is in progress - if users submit a job and after a restart the recovery takes a long time or freezes or fails, they might think their work disappeared although the scheduler seems healthy.  (this is a long reach)